payroll/models.py

```diff
@@ -229,17 +229,17 @@ class EmployerAlias(AliasModel):
     def clean(self):
         super().clean()
 
-        if self.employer.is_department:
-            duplicate_alias = type(self).objects.filter(
-                Q(name=self.name) & Q(employer__parent_id=self.employer.parent.id)
-            )
-        else:
-            duplicate_alias = type(self).objects.filter(
-                Q(name=self.name) & Q(employer__parent_id__isnull=True)
-            )
+        existing_alias = type(self).objects.filter(
+            name=self.name,
+            employer=self.employer
+        ).exclude(
+            id=self.id
+        )
```

Comment:
This prevents creating new aliases that are duplicates, while providing for updates to an existing alias.